PR: https://github.com/odoo/odoo/pull/117667

From: fd74765d7acf86a8f92a1c3927a26a9e3af63a51
From: Adrien Guilliams (adgu)
Date: 2023-04-05 08:45:40

Structural Changes: 5
Total Changes: 10

[IMP] point_of_sale: remove the background loading of partners and products

The loading of products and partners in background is too slow for the moment.
For configurations with a lot of products or partners, the loading is slowing
the whole behaviour of the PoS while it is not really necessary to load
20K products. This commit sets the default behaviour of the PoS to "not load
products and partners in the background", it also adds a button
on the ProductScreen for user to load more products when
a search term is put in the search bar and the same on the PartnerListScreen
for loading partners. This load will load 30 items at a time.

closes odoo/odoo#117667

X-original-commit: 389c85af5983f2f25c7c24f21d3c3716077d7119
Signed-off-by: Trinh Jacky (trj) <trj@odoo.com>
Signed-off-by: Heinz Robin (rhe) <rhe@odoo.com>

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/pos_config.py
+++ b/addons/point_of_sale/models/pos_config.py
@@ -166,14 +166,14 @@ class PosConfig(models.Model):
                                                    "When the session is open, we keep on loading all remaining products in the background.\n"
                                                    "In the meantime, you can click on the 'database icon' in the searchbar to load products from database.")
     limited_products_amount = fields.Integer(default=20000)
-    product_load_background = fields.Boolean(default=True)
+    product_load_background = fields.Boolean(default=False)
     limited_partners_loading = fields.Boolean('Limited Partners Loading',
                                               default=True,
                                               help="By default, 100 partners are loaded.\n"
                                                    "When the session is open, we keep on loading all remaining partners in the background.\n"
                                                    "In the meantime, you can use the 'Load Customers' button to load partners from database.")
     limited_partners_amount = fields.Integer(default=100)
-    partner_load_background = fields.Boolean(default=True)
+    partner_load_background = fields.Boolean(default=False)
     auto_validate_terminal_payment = fields.Boolean(default=True, help="Automatically validates orders paid with a payment terminal.")
     trusted_config_ids = fields.Many2many("pos.config", relation="pos_config_trust_relation", column1="is_trusting",
                                           column2="is_trusted", string="Trusted Point of Sale Configurations",

--- a/addons/point_of_sale/models/res_company.py
+++ b/addons/point_of_sale/models/res_company.py
@@ -7,9 +7,9 @@ class ResCompany(models.Model):
     _inherit = 'res.company'
 
     point_of_sale_update_stock_quantities = fields.Selection([
-            ('closing', 'At the session closing'),
-            ('real', 'In real time (recommended)'),
-            ], default='real', string="Update quantities in stock",
+            ('closing', 'At the session closing (faster)'),
+            ('real', 'In real time (accurate but slower)'),
+            ], default='closing', string="Update quantities in stock",
             help="At the session closing: A picking is created for the entire session when it's closed\n In real time: Each order sent to the server create its own picking")
     point_of_sale_use_ticket_qr_code = fields.Boolean(
         string='Use QR code on ticket',
